tests/integration/plugins/test_npm.py

```diff
@@ -166,3 +183,69 @@ def test_npm_plugin_include_node(new_dir, partitions):
     # try to use bundled Node.js to execute the script
     output = subprocess.check_output([str(node_path), str(binary)], text=True)
     assert output == "hello world\n"
+
+
+@pytest.mark.slow
+def test_npm_plugin_get_file_list(create_fake_package_with_node, new_dir, partitions):
+    parts = create_fake_package_with_node()
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_npm_plugin",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_file_list()
+
+    # This example bundles in node, which brings a ton of other dependencies -
```

Comment:
This is fine but I think this feature would really benefit from a smaller unit test that creates a fake, smaller file tree and then matches the contents of get_file_list() exactly

Reply:
Ok will do

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/898e4ed53674c7caa5246b6d767ccbac1c066a51